weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/traceServerClient.ts

```diff
@@ -60,20 +70,25 @@ export class TraceServerClient extends DirectTraceServerClient {
   }
   public registerOnFeedbackListener(
     weaveRef: string,
-    callback: () => void
+    callback: () => void,
+    feedbackType?: string
   ): () => void {
-    if (!(weaveRef in this.onFeedbackListeners)) {
-      this.onFeedbackListeners[weaveRef] = [];
+    const feedbackTypeResolved = feedbackType ?? this.FEEDBACK_TYPE_DEFAULT;
+    if (!(feedbackTypeResolved in this.onFeedbackListeners)) {
+      this.onFeedbackListeners[feedbackTypeResolved] = {};
+    }
+    if (!(weaveRef in this.onFeedbackListeners[feedbackTypeResolved])) {
+      this.onFeedbackListeners[feedbackTypeResolved][weaveRef] = [];
```

Comment:
your comment says it is going to be `// weave_ref -> feedback_type -> callback`. i think that comment is better than the reverse